src/pages/UserRenewal/UserRenewal.tsx

```diff
@@ -0,0 +1,253 @@
+import { getCities } from "../../api/citiesApi";
+import { NotificationType } from "../../api/NotificationBoxApi";
+import { showUserFormerInfoModal } from "./UserRenewalModals";
+import { useState, useEffect } from "react";
+import Api from "../../api/api";
+import City from "./Types/CityForRenewal";
+import NotificationBoxApi from "../../api/NotificationBoxApi";
+import notificationLogic from "../../components/Notifications/Notification";
+import React from "react";
+import Spinner from "../Spinner/Spinner";
+import styles from "../UserRenewal/UserRenewal.module.css";
+import UserRenewal from "./Types/UserRenewal";
+import userRenewalsApi from "../../api/userRenewalsApi";
+
+const { Step } = Steps;
+
+const timeElapsed = Date.now();
+const today = new Date(timeElapsed);
+
+export default function () {
+    const [form] = Form.useForm();
+    const [loading, setLoading] = useState(false);
+    const [current, setCurrent] = useState(0);
+    const [userId, setUserId] = useState('');
+    const [email, setEmail] = useState('');
+    const [cities, setCities] = useState<City[]>([]);
+    const [notificationTypes, setNotificationTypes] = useState<NotificationType[]>([]);
+    const [cityId, setCityId] = useState(0);
+    const [adminsIds, setAdminsIds] = useState<string[]>([]);
+    const renewal: UserRenewal = ({
+        id: 0,
+        userId: userId,
+        cityId: cityId,
+        requestDate: today,
+        approved: false
+    });
+
+    const next = () => {
+        setCurrent(current + 1);
+    };
+    
+    const validationSchema = {
+        Email: [
+            { required: true, message: emptyInput() },
+            { validator: checkEmail },
+        ]
+    };
+
+    const getNotificationTypes = async () => {
+        setNotificationTypes(await NotificationBoxApi.getAllNotificationTypes());
+    }
+
+    const getCity = async () => {
+        setLoading(true);
+        try {
+            const response = await getCities();
+            let filteredCities = response.data.filter((item: City) => {
+                return item.isActive === true;
+            }).map((data: City) => {
+                return data;
+            });
+            setCities(filteredCities);
+        }
+        finally {
+            setLoading(false);
+        }
+    };
+
+    useEffect(() => {
+        getNotificationTypes();
+        getCity();
+    }, []);
+
+    const checkFormer = async (data: string) => {
+        const response = await userRenewalsApi.checkFormer(data);
+        if (response !== "") {
+            setUserId(response)
+            next();
+        };
+    };
+
+    const handleCheckFormer = async () => {
+        await checkFormer(email);
+    };
+
+    const handleSelectChange = (value: any) => {
+        setCityId(value);
+    };
+
+    const sendRenewal = async (data: any) => {
+        await userRenewalsApi.sendUserRenewal(data)
+            .then(() => {
+                if (adminsIds !== null) {
```

Comment:
If I'm not mistaken, in TS , it's should work just with `if (adminsIds)`

Reply:
Fixed that.